src/runtime/composables/local/useAuth.ts

```diff
@@ -129,11 +129,22 @@ async function getSession(getSessionOptions?: GetSessionOptions): Promise<Sessio
 
   const headers = new Headers(token ? { [config.token.headerName]: token } as HeadersInit : undefined)
 
+  const sessionCookie = useCookie<object | null>('auth:sessionCookie', {
+    default: () => null,
+    maxAge: config.token.maxAgeInSeconds,
+    sameSite: config.token.sameSiteAttribute
+  })
+
   loading.value = true
   try {
     const result = await _fetch<any>(nuxt, path, { method, headers })
     const { dataResponsePointer: sessionDataResponsePointer } = config.session
     data.value = jsonPointerGet<SessionData>(result, sessionDataResponsePointer)
+
+    sessionCookie.value = {
+      lastRefreshedAt: lastRefreshedAt.value,
+      data: data.value
+    }
```

Comment:
Storing user data in the cookie may lead to a GDPR violation if the `getSession` endpoint returns any personal information (which is often the case). It undoubtedly spawns the "protecting personal data" requirement for anyone using the module, even if they weren't affected by the SSG issue in the first place.

https://gdpr.eu/cookies/

I don't find it very attractive to introduce such a burden to NuxtAuth :/

Reply:
Maybe introducing an option to use SessionStorage nor LocalStorage would be great instead of using cookies ?

Reply:
If an attacker gains access to a token stored in a Cookie, Session, or LocalStorage, the primary concern is not just identifying the user, but also the ability to access personal data and perform actions the user is authorized to do. This presents a clear security risk.

From a GDPR perspective, the critical question is whether the token can be used to track the user or if it contains personal data. Regardless of where the token is stored, if it can uniquely identify a user or track their actions, it falls under GDPR regulations.

For example, JWTs often store a user identifier in the payload. Since JWTs are not opaque and can be decoded, they may be used to directly identify a user, making them subject to GDPR. Even if the token does not explicitly store "personal data" like an email address, it qualifies as personal data if it can uniquely identify a user (e.g., through a user ID).

In this case, you must inform users via a cookie notice or a similar consent mechanism, ensuring transparency about the use of the token and obtaining their consent where required. However, if the cookies are strictly necessary for the operation of the web application (e.g., to keep users logged in or enable certain features), explicit consent may not be required under GDPR. This generally applies to authentication cookies, afaik.

Reply:
@thorge You are mostly right. However, auth cookies/jwts don't need to store emails or other identifying information. A website can use "shallow" JWTs storing only user id in it (this is a valid `Access Token` in NuxtAuth terms). As to whether an ID is "personal data" is very subjective.

However if a website uses such a "shallow" JWT (`Access Token`) and makes a call to a `getSession` endpoint to check its validity + get user data, then it's a different thing. For example, internally we optimize such a call to also return user email, name, permissions, etc. on top of a regular JWT validity check. I would prefer to not store this information in a cookie.

Please note that I am also not clearly understanding the usecase of static websites fully. If there is authentication, what is the issue with the session being refetched on page reload?

Reply:
@phoenix-ru There’s actually no need to store the session itself; we only need to store the access token. That’s what I thought this was about, but I must admit I didn’t examine the changes in detail. Sorry for my tldr answer above :-)

As @Geekimo and @Vijayabhaskar96 suggested, I’d prefer a solution where storing the access token in a cookie (or alternatively in SessionStorage/LocalStorage) is optional. If the session data isn’t available after a client page reload, the access token could be used to retrieve the session data as needed.